frontend/src/modules/fluenteditor/components/Editor.js

```diff
@@ -153,27 +145,32 @@ export class EditorBase extends React.Component<EditorProps, State> {
     }
 
     render() {
-        let EditorImplementation = SourceEditor;
-        if (!this.state.forceSource && this.state.syntaxType === 'simple') {
+        let EditorImplementation = RichEditor;
+        if (this.state.forceSource || this.state.syntaxType === 'complex') {
+            EditorImplementation = SourceEditor
+        }
+        else if (this.state.syntaxType === 'simple') {
             EditorImplementation = SimpleEditor;
         }
 
-        // If the type of the string is not 'complex', meaning we show an
-        // optimized editor by default, show a button to allow switching to
-        // the source editor.
-        const ftlSwitch = this.state.syntaxType === 'complex' ? <button
-            className='ftl active'
-            title='Advanced FTL mode enabled'
-            onClick={ this.props.showNotSupportedMessage }
-        >
-            FTL
-        </button> : <button
-            className={ 'ftl' + (this.state.forceSource ? ' active' : '') }
-            title='Toggle between simple and advanced FTL mode'
-            onClick={ this.toggleForceSource }
-        >
-            FTL
-        </button>;
+        // Show a button to allow switching to the source editor.
+        let ftlSwitch = null;
+        // But only if the user is logged in and the string is not read-only.
+        if (this.props.user.isAuthenticated && !this.props.isReadOnlyEditor) {
+            ftlSwitch = this.state.syntaxType === 'complex' ? <button
+                className='ftl active'
+                title='Advanced FTL mode enabled'
+                onClick={ this.props.showNotSupportedMessage }
+                >
```

Comment:
I'm not sure if this is indented properly or not. I'll let you make the call.

Reply:
It isn't!